### PR TITLE
CHOPIN-43961: Dependency upgrade

### PR DESCRIPTION
**Pull request by bredy**
- migreted from Spring Boot 1.x to 2.x
- migrated from Saxon-HE 9.7 to 10.2
- minor security hardening

### CODE REVIEW

File: src/test/java/com/cisco/cta/taxii/adapter/TcpServer.java [hunk 1]
```diff
@@ -27,7 +27,7 @@
 
 class TcpServer extends Thread implements AutoCloseable {
 
-    private static int port = 9000;
+    private static int port = 19000;
```
Comment (MikkCZ) [inline src/test/java/com/cisco/cta/taxii/adapter/TcpServer.java:30]: Why changing the port?

File: src/main/java/com/cisco/cta/taxii/adapter/AdapterConfiguration.java [hunk 2]
```diff
@@ -99,7 +94,10 @@ public TaxiiPollResponseReaderFactory taxiiPollResponseReaderFactory() throws Ex
 
     @Bean
     public XMLInputFactory inputFactory() {
-        return XMLInputFactory.newFactory();
+        XMLInputFactory xmlInputFactory = XMLInputFactory.newFactory();
```
Comment (MikkCZ) [inline src/main/java/com/cisco/cta/taxii/adapter/AdapterConfiguration.java:97]: I see you added protection against XXE. 👍 for that. Can you please also add a unit test to verify both the `XMLInputFactory` and `TransformerFactory` are actually resistant to such attack? Here is an example of the potentially malicious payload. If encountered, the deserialization process should fail.

```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<!DOCTYPE Collection_Information_Request [
        <!ENTITY foo "bar">
        ]>
<taxii_11: ...
```

**Review by MikkCZ — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -1 +1,29 @@
...
+#### RELEASE NOTES
+
+##### Version 2.0.0
+
+* Due to migration to Spring Boot 2.0 we renamed configuration property `taxiiService` to `taxii-service`
+  because camel case configuration properties are no longer supported.
+
+* Slash character `/` in JSON output is now escaped by default and is returned as `\/` according to changes in Saxon-HE 9.7.0-8.
+  When you'd like to keep backward compatibility you need to modify stylesheet and use e.g. character maps as shown in
+  following example:
```
Comment (mvelkCisco) [inline README.md:12]: Slash character `/` in JSON output is now escaped by default and it's returned as `\/` according to the changes in Saxon-HE 9.7.0-8.
If you'd like to keep backward compatibility you can modify the stylesheet and use character maps as shown in the  following example:

File: README.md [hunk 4]
```diff
@@ -1 +1,29 @@
...
+    <xsl:variable name="serialize-options" as="map(xs:string, xs:anyAtomicType)">
+        <xsl:map>
+            <xsl:map-entry key="'method'" select="'json'"/>
+            <xsl:map-entry key="'use-character-maps'" select="$character-map"/>
+        </xsl:map>
+    </xsl:variable>
+
+    <xsl:value-of select="f:replace-doubles-ints(fn:serialize(fn:parse-json(fn:xml-to-json($xml)), $serialize-options))"/>
+  ```
+  You need to add also XML namespace support `xmlns:xs="http://www.w3.org/2001/XMLSchema"` if it's not already in stylesheet.
```
Comment (mvelkCisco) [inline README.md:29]: You need to also add XML namespace support `xmlns:xs="http://www.w3.org/2001/XMLSchema"` if it's not already in the stylesheet.

**Review by mvelkCisco — CHANGES_REQUESTED** [review]

File: src/test/java/com/cisco/cta/taxii/adapter/TcpServer.java (hunk 1 above)
Comment (bredy) [reply src/test/java/com/cisco/cta/taxii/adapter/TcpServer.java:30]: Beacuse I had conflict with my localhost Sonar server and suppose testing server could have port > 10000 not to interfere with common services.

**Review by MikkCZ — APPROVED** [review]

**Review by mvelkCisco — APPROVED** [review]

**Pull request merged**